### PR TITLE
feat(config): docs.excludeModules — doc-only module exclusion independent of activation

### LINKED ISSUE #3923: 🔧 config.docs.excludeModules — decouple doc contribution from module runtime activation
## Problem

The public OpenAPI spec (`GET /api/spec.json`) and the public docs guide tree (`GET /api/public/docs`) are assembled by globbing every module's `doc/*.yml` + `doc/guides/*.md` (`config/assets.js`). That resolved file list is then filtered **only by module runtime activation** (`config/index.js` → `lib/helpers/config.js` `filterByActivation`, which reads `config[moduleName].activated`).

Two coupling defects fall out of this:

1. **Activation conflates runtime with documentation.** A module's `activated` flag gates BOTH its routes/models AND its doc contribution. There is no way to keep a module running while dropping only its sample docs.
2. **Core modules bypass the filter entirely.** `CORE_MODULES = { core, auth, users, home }` are never filtered, because they carry infra-critical endpoints — `home` serves `/api/health` (liveness) and `/api/admin/readiness`. So a core module's sample doc/guide content (`home/doc/guides/00-welcome.md` → slug `welcome`, `01-quickstart.md` → slug `quickstart`) is **always** globbed into the tree, with no opt-out.

### Impact

A consumer that ships its own onboarding guides under the natural slugs `welcome` / `quickstart` collides with the `home` samples → the docs tree builder rejects duplicate slugs. The only workaround is to physically delete the sample guides shipped by the framework — which conflicts with keeping framework files byte-identical across upgrades and so recurs on every sync.

(The sample OpenAPI doc `tasks/doc/tasks.yml` is **not** affected — `tasks` is non-core, so `config.tasks.activated = false` already strips it via the existing mechanism. This issue is specifically the core-module doc case that activation cannot reach.)

## Proposed fix

Add an **independent doc-exclusion** config: `config.docs.excludeModules: string[]` (default `[]`). A new helper `filterByDocExclusion(files, config)` drops `doc/*.yml` (spec) + `doc/guides/*.md` (guides) for any module in the list — **regardless of activation / core status** — applied as a second filter pass over the `openapi` + `guides` file keys, after `filterByActivation`. Default empty keeps the sample guides as a working tutorial.

This decouples documentation contribution from runtime activation (doc ≠ runtime), is granular per-module, and works for core and non-core modules alike.

## Scope

- `lib/helpers/config.js`: add `filterByDocExclusion(files, config)` (reuse `extractModuleName`; **no `CORE_MODULES` bypass**; missing/empty list → no-op).
- `config/index.js`: second filter pass over `['openapi', 'guides']` after the `filterByActivation` loop. Runtime file keys (routes/models/policies/configs/preRoutes) untouched.
- `config/defaults/development.config.js`: `docs.excludeModules: []` + comment (doc-only exclusion, independent of activation, works on core modules; default empty keeps samples).
- Unit tests: excludes a listed module's `doc/*.yml` + `doc/guides/*.md`; keeps non-listed modules; **works for a core module** (the motivating regression); default `[]` / missing = no-op; runtime file keys unaffected by `excludeModules`.
- `MIGRATIONS.md` entry: new `config.docs.excludeModules` knob + when a consumer would set it (a runtime-active module whose sample docs collide with the consumer's own).

## Out of scope

- `tasks/doc/tasks.yml` (already handled by `config.tasks.activated = false`).
- Any tree-builder dedup/precedence change.

Created via /dev:issue

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## Summary

Adds `config.docs.excludeModules` (default `[]`) — a **doc-only** filter that drops a module's `doc/*.yml` (OpenAPI) + `doc/guides/*.md` from the public spec (`/api/spec.json`) and guide tree (`/api/public/docs`), **independent of module runtime activation**. It works even for core modules (`core`/`auth`/`users`/`home`), which `filterByActivation` never filters.

This lets a project keep a module runtime-active (e.g. `home` serves `/api/health`) while dropping only its sample docs/guides — avoiding a guide-slug collision with the project's own guides. Previously a module's `activated` flag gated both runtime and docs, and core modules bypassed the filter entirely, so their sample docs/guides were always served with no opt-out.

## Changes

- `lib/helpers/config.js` — new `filterByDocExclusion(files, config)` next to `filterByActivation` (**no `CORE_MODULES` bypass**; missing/empty/non-array list = no-op) + default export.
- `config/index.js` — second filter pass over the `openapi` + `guides` file keys, after `filterByActivation`. Runtime file keys (routes/models/policies/...) are unaffected.
- `config/defaults/development.config.js` — `docs.excludeModules: []` (no behavior change).
- Unit tests (9, incl. core-module + no-op + exact-match cases) + `MIGRATIONS.md` entry.

## Testing

- `npm run lint` — 0 issues.
- Unit suite green — 2102 tests (incl. 9 new).
- Default `[]` = no-op → zero behavior change for existing deployments.

Closes #3923

https://claude.ai/code/session_018zwvrFqZM1y68TWA8oVW3n

## Summary by CodeRabbit

* **New Features**
  * Added an optional setting to exclude selected modules’ documentation from public API docs and the OpenAPI spec.
  * Documentation files for excluded modules are now hidden even when the module itself is still active.
  * Updated the migration notes with setup guidance and an example configuration.

* **Tests**
  * Added coverage for module-based documentation exclusion behavior and edge cases.

### CODE REVIEW

File: lib/helpers/config.js [hunk 1]
```diff
@@ -203,6 +203,35 @@ const filterByActivation = (files, config) => files.filter((file) => {
...
+ */
+const filterByDocExclusion = (files, config) => {
+  const excluded = config.docs?.excludeModules;
+  if (!Array.isArray(excluded) || excluded.length === 0) return files; // no-op
+  const excludedSet = new Set(excluded);
+  return files.filter((file) => {
+    const moduleName = extractModuleName(file);
+    if (!moduleName) return true; // not a module file, keep it
+    return !excludedSet.has(moduleName); // drop listed modules (incl. core)
+  });
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply lib/helpers/config.js:235]: Addressed in bddff919 — added a `/doc/` path guard so the helper only ever drops files under a module's `doc/` directory, matching the doc-only contract. Runtime files are never removed even if called with a broader list than the `openapi`+`guides` keys it's wired on. Added a unit test covering it.

File: config/index.js [hunk 2]
```diff
@@ -164,6 +164,15 @@ const initGlobalConfig = async () => {
...
   }
+  // Exclude doc files (OpenAPI YAML + guides) for modules listed in
+  // config.docs.excludeModules — independent of runtime activation, so it works
+  // even for core modules. Empty/missing list = no-op.
+  const docFileKeys = ['openapi', 'guides'];
+  for (const key of docFileKeys) {
+    if (config.files[key]) {
+      config.files[key] = configHelper.filterByDocExclusion(config.files[key], config);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply config/index.js:175]: `initGlobalConfig` already has a JSDoc header at config/index.js L87-91 ("Initialize global configuration by layering module defaults, global defaults, environment overrides, and DEVKIT_NODE_* env vars" + `@returns`). The flagged L92 is the function declaration; the JSDoc sits immediately above it. No change needed.
